arachni/app/web.py

```diff
@@ -0,0 +1,146 @@
+    ('audit-with-extra-parameter', '--audit-with-extra-parameter'),
+    ('audit-with-both-methods', '--audit-with-both-methods'),
+    ('audit-exclude-vector', '--audit-exclude-vector'),
+    ('audit-include-vector', '--audit-include-vector'),
+    ('http-user-agent', '--http-user-agent'),
+    ('http-request-timeout', '--http-request-timeout'),
+    ('http-request-redirect-limit', '--http-request-redirect-limit'),
+    ('http-request-header', '--http-request-header'),
+    ('http-response-max-size', '--http-response-max-size'),
+    ('http-cookie-string', '--http-cookie-string'),
+    ('http-authentication-username', '--http-authentication-username'),
+    ('http-authentication-password', '--http-authentication-password'),
+    ('http-authentication-type', '--http-authentication-type'),
+    ('http-ssl-verify-peer', '--http-ssl-verify-peer'),
+    ('http-ssl-verify-host', '--http-ssl-verify-host'),
+    ('http-ssl-version', '--http-ssl-version'),
+    ('input-value', '--input-value'),
+    ('input-without-defaults', '--input-without-defaults'),
+    ('input-force', '--input-force'),
+    ('checks-list', '--checks-list'),
+    ('checks', '--checks'),
+    ('plugins-list', '--plugins-list'),
+    ('plugin', '--plugin'),
+    ('platforms-list', '--platforms-list'),
+    ('platforms-no-fingerprinting', '--platforms-no-fingerprinting'),
+    ('session-check-url', '--session-check-url'),
+    ('session-check-pattern', '--session-check-pattern'),
+    ('browser-cluster-ignore-images', '--browser-cluster-ignore-images'),
+    ('browser-cluster-screen-width', '--browser-cluster-screen-width'),
+    ('browser-cluster-screen-height', '--browser-cluster-screen-height'),
+    ('timeout', '--timeout'),
+    ('timeout-suspend', '--timeout-suspend'),
+])
+
+app = Flask(__name__)
+sentry = Sentry(app)
+
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    if request.method == 'GET':
+        return render_template('index.html', options=ARGS)
+
+    if 'to' not in request.form or not request.form['to']:
+        abort(400)
+    elif 'website-url' not in request.form or not request.form['website-url']:
+        abort(400)
+
+    email_plugin_args = ['report=txt',
+                         'from={}'.format(SES_FROM),
+                         'bcc={}'.format(ROOT_EMAIL),
+                         'server_address={}'.format(SES_SERVER),
+                         'tls=true',
+                         'server_port={}'.format(SES_PORT),
+                         'username={}'.format(SES_USERNAME),
+                         'password={}'.format(SES_PASSWORD)]
+
+    url = None
+    options = []
+    for k, v in request.form.iteritems():
+        if k == 'qa':
+            email_plugin_args.append('cc={}'.format(QA_EMAIL))
+        elif k == 'to':
+            email_plugin_args.append('to={}'.format(v))
+        elif k == 'website-url':
+            url = v
+        elif k in ARGS:
+            # XXX: since we're using raw user input in shell command
```

Comment:
This is not OK, you can avoid any shellout very easily (see previous comment near `Popen`).